[lunaroddity] iP

lunaroddity opened the pull request:
# Bareum
![4780571b-72e3-41e1-b22d-eee21f3eb4b9](https://github.com/nus-cs2103-AY2324S1/ip/assets/101786124/1c5cfa9b-cd0a-4389-9c0f-cdb0667b2538)
> _If you can't tell how near or far it is, why don't you start with the things that are within reach?
-Doeun_

🕙💯 Always organised and on top of things, Bareum **never** forgets a task! Using a text-based interface, it's
- light
- easy to learn
- easy to use!

Try out Bareum for **FREE** [here](https://github.com/lunaroddity/ip)! Just
1. Download it
2. Run it
3. Add your tasks!

## Features
- [x] Stay on top of things by letting Bareum manage your tasks, deadlines and events!
- [ ] Find a text-based interface hard-to-read? Use the Graphical User Interface (GUI) version! (coming soon)

### Like Java?
Bareum does too! ~~of the drinkable variety~~
Here's the main method!
```java
public static void main(String[] args) {
        new Bareum("./data/storedTasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

lunaroddity committed:
Rename chatbot to Bareum, add greetings

lunaroddity committed:
Add echo command, add exit when user types bye

lunaroddity committed:
Add TaskList and Task classes, add add task and list tasks features

lunaroddity committed:
Add mark as done and mark as undone features

lunaroddity committed:
Add ToDos, Events and Deadlines types

lunaroddity committed:
Add Text UI Testing

lunaroddity committed:
Add error handling for all task types

lunaroddity committed:
Add delete task feature

lunaroddity committed:
Add save feature

lunaroddity committed:
Merge pull request #1 from lunaroddity/branch-Level-7

Add save feature

lunaroddity committed:
Change DeadlineTask dates to java.time.LocalDate

lunaroddity committed:
Merge pull request #2 from lunaroddity/branch-Level-8

Change DeadlineTask dates to java.time.LocalDate

lunaroddity committed:
Issue reproduction

lunaroddity committed:
Add Command, Ui, Parser and Storage classes

lunaroddity committed:
Merge pull request #3 from lunaroddity/branch-More-OOP

More OOP

lunaroddity committed:
Add bareum and bareum.commands package

lunaroddity committed:
Merge pull request #4 from lunaroddity/branch-Packages

Add bareum and bareum.commands package

lunaroddity committed:
Merge remote-tracking branch 'origin/add-gradle-support'

lipwei1808 commented on src/main/java/bareum/Parser.java:
```diff
@@ -0,0 +1,48 @@
+import bareum.commands.UnmarkCommand;
+
+public class Parser {
+    public Parser() {
+
+    }
```
Maybe you forgot to remove this?

lunaroddity replied:
Wasn't really sure if I needed a constructor if it was empty so I created one just in case but in retrospect, it's redundant so I'll remove it

lipwei1808 commented on src/main/java/bareum/Storage.java:
```diff
@@ -0,0 +1,81 @@
+        }
+        this.storedTasks = storedTasks;
+        StringBuilder savedTasks = new StringBuilder();
+        int currChar;
```
Maybe the naming could be better? Since it's an integer

lunaroddity replied:
Thanks for the feedback! I'll probably change it to something like `int currCharAsciiCode`

lipwei1808 commented on src/main/java/bareum/Storage.java:
```diff
@@ -0,0 +1,81 @@
+                savedTasks.append((char) currChar);
+            }
+
+            String[] allTaskInputs = savedTasks.toString(). split("\n");
```
There is an extra space before the split method

lipwei1808 commented on src/main/java/bareum/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,36 @@
+    public AddDeadlineCommand(String[] commandInputs) throws BareumException {
+        if (commandInputs.length <= 1) {
+            throw new BareumException("You are missing some arguments :(\n" +
+                    "Current number of arguments: " + commandInputs.length +
```
Should leave a new line after every "+"

lipwei1808 commented on src/main/java/bareum/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,36 @@
+            DeadlineTask task = DeadlineTask.makeDeadline(allDetails);
+            taskList.addTask(task);
+            storage.saveNewTask(task);
+            String added = "I have added this task:\n" + task + "\nYou now have "
+                    + taskList.size() + " task(s) in your list.";
+            Ui.reply(added);
```
Same as above

lipwei1808 commented on src/main/java/bareum/DeadlineTask.java:
```diff
@@ -0,0 +1,61 @@
+    public static DeadlineTask makeDeadline(String allDetails) throws BareumException {
+        if (allDetails.length() == 0) {
+            throw new BareumException("Deadline description cannot be empty.");
+
```
You got an extra line here

lipwei1808 commented on src/main/java/bareum/DeadlineTask.java:
```diff
@@ -0,0 +1,61 @@
+        super(isDone, description);
+        this.dueDate = dueDate;
+        this.isDone = false;
+        // add exceptions for time
```
I think this is for a future addition right or is this misplaced

lunaroddity replied:
Yep, I planned to add this in a future commit and left this as a reminder to myself :") Thanks for the feedback!

lunaroddity committed:
Add JUnit tests

lunaroddity committed:
Merge pull request #5 from lunaroddity/branch-JUnit

Add JUnit tests

lunaroddity committed:
Change shadowJar archiveBaseName and add archiveFileName

lunaroddity committed:
Merge pull request #6 from lunaroddity/branch-Jar

Change shadowJar archiveBaseName and add archiveFileName

lunaroddity committed:
Add JavaDocs to all methods and classes

lunaroddity committed:
Adhere to coding standard

lunaroddity committed:
Add find feature

lunaroddity committed:
Merge pull request #7 from lunaroddity/branch-A-JavaDoc

Add JavaDocs to all methods and classes

lunaroddity committed:
Merge branch 'master' into branch-A-CodingStandard

lunaroddity committed:
Merge pull request #8 from lunaroddity/branch-A-CodingStandard

Adhere to coding standard

lunaroddity committed:
Merge branch 'master' into branch-Level-9

lunaroddity committed:
Merge pull request #9 from lunaroddity/branch-Level-9

Add find feature

lunaroddity committed:
Add checkstyle and checkstyle plugin to build.gradle

lunaroddity committed:
Merge pull request #10 from lunaroddity/branch-A-Checkstyle

Add checkstyle and checkstyle plugin to build.gradle

ryanongwx commented on src/test/java/bareum/ParserTest.java:
Great. I feel that your test have encompass most of the scenarios that your chatbot will be put through

ryanongwx commented on src/main/java/bareum/EventTask.java:
Your functions are well commented such that readers will immediately be able to understand your program

ryanongwx commented on src/main/java/bareum/Ui.java:
There is a redundant space which changes the indent to 5 spaces instead of the conventional 4 spaces

ryanongwx commented on src/main/java/bareum/commands/AddTodoCommand.java:
Good job keeping fields private

lunaroddity committed:
Add GUI, test for FindCommand

lunaroddity committed:
Merge pull request #11 from lunaroddity/branch-Level-10

Add GUI, test for FindCommand

lunaroddity committed:
Add assertions for DeadlineTask and EventTask

The private constructor methods in DeadlineTask and EventTask assume
that the times and dates passed to them are non-null. Using assertions
to ensure that this is true and confirms these assumptions, making the
program less error-prone.

lunaroddity committed:
Improve code quality

The Parser.parse() command was too lengthy and hard-to-read.
The parsing of individual commands was abstracted to shorten it.
SLAP was implemented, hence shorter commands like 'bye' and 'list'
were also abstracted.

lunaroddity committed:
Merge pull request #13 from lunaroddity/branch-A-CodeQuality

Improve code quality

lunaroddity committed:
Merge pull request #14 from lunaroddity/branch-A-Assertions

Add assertions for DeadlineTask and EventTask

lunaroddity committed:
Add tagging feature

lunaroddity committed:
Merge pull request #15 from lunaroddity/branch-C-Tagging

Add tagging feature

lunaroddity committed:
Add tests for parsing tag commands from user in Parser.parse()

lunaroddity committed:
Fix extra whitespace before tag, fix todo bug when loading from save

lunaroddity committed:
Merge pull request #16 from lunaroddity/branch-C-Tagging

lunaroddity committed:
Add user guide and UI screenshot

lunaroddity committed:
Merge remote-tracking branch 'origin/master'

lunaroddity committed:
Add user guide

lunaroddity committed:
Merge remote-tracking branch 'origin/master'

lunaroddity committed:
Update README.md

lunaroddity committed:
Update README.md

lunaroddity committed:
Update README.md

lunaroddity committed:
Update README.md

lunaroddity committed:
Update README.md

lunaroddity committed:
Update README.md

lunaroddity committed:
Update README.md
